proxy/http/HttpTransact.cc

```diff
@@ -1826,10 +1830,14 @@ HttpTransact::ReDNSRoundRobin(State *s)
     s->next_action = how_to_open_connection(s);
   } else {
     // Our ReDNS failed so output the DNS failure error message
-    build_error_response(s, HTTP_STATUS_BAD_GATEWAY, "Cannot find server.", "connect#dns_failed");
+    // Set to internal server error so later logging will pick up SQUID_LOG_ERR_DNS_FAIL
```

Comment:
Not 100% sure if we can do this separately, but it doesn't seem like a change for H2 to Origin.

Reply:
This is actually part of PR #7580 as is the line with your previous comment.  I'm hoping it lands first and will rebase out of this PR.  Removing it makes my rebase back to a prod build more sketchy.  However, I've had no luck getting anyone to look at it.